model/sparql/GenericSparql.php

```diff
@@ -2229,19 +2228,28 @@ public function listConceptGroupContents($groupClass, $group, $lang,$showDepreca
      * @param string $lang language of labels to return
      * @param int $offset offset of results to retrieve; 0 for beginning of list
      * @param int $limit maximum number of results to return
+     * @param boolean $showDeprecated whether to include deprecated concepts in the change list
      * @return string sparql query
      */
-    private function generateChangeListQuery($prop, $lang, $offset, $limit=200) {
+    private function generateChangeListQuery($prop, $lang, $offset, $limit=200, $showDeprecated) {
         $fcl = $this->generateFromClause();
         $offset = ($offset) ? 'OFFSET ' . $offset : '';
 
+        //Additional clauses when deprecated concepts need to be included in the results
+        $optionVar = ($showDeprecated) ? '?replacedBy' : '';
+        $optionStart = ($showDeprecated) ? 'OPTIONAL {' : '';
+        $optionEnd = ($showDeprecated) ? '}' : '';
+        $deprecatedOption = ($showDeprecated) ? 'OPTIONAL { ?concept dc:modified ?date ; owl:deprecated TRUE ; dc:isReplacedBy ?replacedBy }' : '';
+
         $query = <<<EOQ
-SELECT DISTINCT ?concept ?date ?label $fcl
+SELECT DISTINCT ?concept ?date ?label $optionVar $fcl
 WHERE {
   ?concept a skos:Concept .
-  ?concept $prop ?date .
   ?concept skos:prefLabel ?label .
   FILTER (langMatches(lang(?label), '$lang'))
+  $optionStart ?concept $prop ?date .
+  MINUS{ ?concept owl:deprecated TRUE } $optionEnd
```

Comment:
OPTIONAL + MINUS some of the very same matches does not seem right and may be the cause of performance regressions, especially combined with another OPTIONAL (OPTIONALs are not performance-friendly, at least not on Fuseki based on my own research). Please consider alternative approach, e.g., inner UNIONs can, in many cases, be applied instead and may even simplify the code. It should be an easy task for the triple store to retrieve the labels when the query is optimally constructed.

(Another performance-wise point regarding the whole code base: we should change from using the FROM clause into GRAPH based filtering.)